### PR TITLE
http default user

### PR DESCRIPTION
**Pull request by chaudum**

### CODE REVIEW

File: sql/src/main/java/io/crate/rest/action/RestSQLAction.java [hunk 1]
```diff
@@ -114,10 +111,19 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
         return Option.NONE;
     }
 
+    @VisibleForTesting
+    String userFromRequest(RestRequest request) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/rest/action/RestSQLAction.java]: can be static

File: sql/src/test/java/io/crate/rest/action/RestSQLActionTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.util.Collections;
+
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Mockito.mock;
+
+public class RestSQLActionTest extends CrateUnitTest {
+
+    @Test
+    public void testUserFromHttpRequest() throws Exception {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/rest/action/RestSQLActionTest.java]: can we split this into multiple tests? - e.g. testDefaultUserIfNoUserHeaderIsPresent, etc..

File: sql/src/main/java/io/crate/rest/action/RestSQLAction.java (hunk 1 above)
Comment (chaudum) [inline sql/src/main/java/io/crate/rest/action/RestSQLAction.java]: so you would rather pass the settings as argument than using the settings of the class?

File: sql/src/test/java/io/crate/rest/action/RestSQLActionTest.java (hunk 2 above)
Comment (chaudum) [inline sql/src/test/java/io/crate/rest/action/RestSQLActionTest.java]: sure

File: sql/src/main/java/io/crate/rest/action/RestSQLAction.java (hunk 1 above)
Comment (mfussenegger) [inline sql/src/main/java/io/crate/rest/action/RestSQLAction.java]: ah I missed the settings. Never mind leave it as is.

**Pull request merged**